app/src/main/java/com/example/poste/activities/EditPostActivity.java

```diff
@@ -59,84 +39,73 @@ public class EditPostActivity extends AppCompatActivity {
+        Folder currentFolder = User.getUser().getFolder(folderId);
+        if (currentFolder == null){
+            currentPost = null;
+            Log.e("Error", "onCreate: Folder not found");
+        } else {
+            currentPost = User.getUser().getFolder(folderId).getPostFromFolder(postId);
+            if (currentPost == null){
+                Log.e("Error", "onCreate: Post not found");
+            } else {
+                postTitle.setText(currentPost.getTitle());
+                postDescription.setText(currentPost.getDescription());
+                linkToPost.setText(currentPost.getUrl());
             }
-        } catch (APIException e) {
-            throw new RuntimeException(e);
         }
 
-        // Setup for the dropdown menu
-        ArrayAdapter<String> adapter = new ArrayAdapter<String>(this, android.R.layout.simple_spinner_item, new ArrayList<>(folderIdNameMap.keySet()));
-        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
-        spinner.setAdapter(adapter);
-
-        // Set the default values
-        postNameField.setText(currentPost.getName());
-        postLinkField.setText(currentPost.getLink());
-        spinner.setSelection(adapter.getPosition(currentFolderTitle));
-
-        // Cancel button click handler
-        cancelBtn.setOnClickListener(view -> {
-            // Send back to folder view
-            sendToDashboard();
+        buttonCancelChanges.setOnClickListener(view -> {
+            Intent returnToFolderIntent = new Intent(EditPostActivity.this, EditPostActivity.class);
+            finish();
+            returnToFolderIntent.putExtra("postID",postId);
+            returnToFolderIntent.putExtra("folderID",folderId);
+            startActivity(returnToFolderIntent);
         });
-
-        // Save button click handler
-        saveBtn.setOnClickListener(view -> {
-            try {
-                String selectedFolderName = spinner.getSelectedItem().toString();
-                Integer selectedFolderId = folderIdNameMap.get(selectedFolderName);
-
-                // Save the post in the API
-                if (currentPost.getName() != postNameField.getText().toString() ||
-                        currentPost.getLink() != postLinkField.getText().toString()) {
-                    API.updatePost(currentPost.getId(), postNameField.getText().toString(), postLinkField.getText().toString(), currentPost.getOwnerId());
-                }
-
-                // Update the post in the API
-                if (currentFolder.getId() != selectedFolderId) {
-                    API.removePostFromFolder(currentPost.getId(), currentFolder.getId());
-                    API.addPostToFolder(currentPost.getId(), selectedFolderId);
-                }
-
-                // Send back to folder view
-                sendToDashboard();
-            } catch (APIException e) {
-                throw new RuntimeException(e);
+        buttonSaveChanges.setOnClickListener(view -> {
```

Comment:
is there a way to exit the activity?

Reply:
Cancel should take me back to the folder. I wasn't sure how to do that without the app crashing since the folder refactor hadnt been done yet. I'll clean that up and test it and commit again